contracts/V1/StateChannelDiamondProxy/DisputeManagerFacet.sol

```diff
@@ -14,26 +14,28 @@ contract DisputeManagerFacet is StateChannelCommon {
         bytes memory signature
     ) public { 
         // sanity checks
-        msg.sender == dispute.disputer;
+        require(msg.sender == dispute.disputer, "CREATE DISPUTE: INVALID DISPUTER");
         // race condition check
-        if(onChainSlashes.length != onChainSlashedParticipants.length 
-        || onChainSlashes[onChainSlashes.length - 1] != onChainSlashedParticipants[onChainSlashedParticipants.length - 1]) {
-            revert("onChainSlashes does not match onChainSlashedParticipants");
+        address[] memory onChainSlashedParticipants = getOnChainSlashedParticipants();
+        if(dispute.onChainSlashes.length != onChainSlashedParticipants.length 
+        || dispute.onChainSlashes[dispute.onChainSlashes.length - 1] != onChainSlashedParticipants[onChainSlashedParticipants.length - 1]) {
```

Comment:
My original comment on this still holds - you're checking just the last element to be equal instead of the entire array + if both arrays are empty this will break since you're accessing the array at position -1

Either iterate and compare or hash the array and compare

Reply:
Okey so my idea was this, to save gas, the last element will be a commitment of all element inside hash(all elements), so if its equal meaning we verified all elements and the issue of accessing empty array yaap am fixing it

Reply:
This could work, since onChainSlashed is append only and only cleared on successful submission of the snapshot. So if a peer knows the K-th element it also knows all the K-1 elements prior and since the past elements can't change, this could also work. Just implement it without the [-1] access